bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/ui/launch/shortcuts/StartAction.java

```diff
@@ -99,42 +98,16 @@ public void launch(IEditorPart part, String mode) {
      * 
      * @throws Exception
      */
-    public static void run(IProject iProject, ILaunchConfiguration iConfiguration, String mode) throws Exception {
-
-        if (Trace.isEnabled()) {
-            Trace.getTracer().traceEntry(Trace.TRACE_UI, new Object[] { iProject, iConfiguration, mode });
-        }
-
-        if (iProject == null) {
-            throw new Exception("Invalid project. Be sure to select a project first.");
-        }
+    public static void run(IProject iProject, String mode) throws Exception {
 
         // Validate that the project is supported.
         DevModeOperations devModeOps = DevModeOperations.getInstance();
         devModeOps.verifyProjectSupport(iProject);
 
-        // If the configuration was not provided by the caller, determine what configuration to use.
+        // Determine what configuration to use.
         LaunchConfigurationHelper launchConfigHelper = LaunchConfigurationHelper.getInstance();
-        ILaunchConfiguration configuration = (iConfiguration != null) ? iConfiguration
-                : launchConfigHelper.getLaunchConfiguration(iProject, mode, RuntimeEnv.LOCAL);
-
-        // Save the time when this configuration was processed.
-        launchConfigHelper.saveConfigProcessingTime(configuration);
+        ILaunchConfiguration configuration = launchConfigHelper.getLaunchConfiguration(iProject, mode, RuntimeEnv.UNKNOWN);
 
-        // Retrieve configuration data.
-        boolean runInContainer = configuration.getAttribute(StartTab.PROJECT_RUN_IN_CONTAINER, false);
-        String configParms = configuration.getAttribute(StartTab.PROJECT_START_PARM, (String) null);
-        String javaHomePath = JRETab.resolveJavaHome(configuration);
-
-        // Process the action.
-        if (runInContainer) {
-            devModeOps.startInContainer(iProject, configParms, javaHomePath, mode);
-        } else {
-            devModeOps.start(iProject, configParms, javaHomePath, mode);
-        }
-
-        if (Trace.isEnabled()) {
-            Trace.getTracer().traceExit(Trace.TRACE_UI);
-        }
+        DebugUITools.launch(configuration, mode);
```

Comment:
OK, it took me a bit but I think I get the refactoring here.

It was previously a bit weird how'd we call back through StartAction from LaunchConfigurationDelegateLauncher, even in cases where we might be doing devc (container).

I think this is overall a nicer separation and flow.

I don't think you did it purely for clean code though.  I think you did it so that this `DebugUITools.launch` always gets called, so our LaunchConfigurationDelegateLauncher always gets called and passed an ILaunch impl.

I think that's better.   We have some decent test coverage so hopefully can feel fairly good that everything necessary has been moved.